app/js/enhancedDate.js

```diff
@@ -1,5 +1,118 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var myDate = (function iife() {    
+    'use strict';
 
+    var theDate;
+    var now;
+    var dayNames = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+        'Friday', 'Saturday'];
+    var monthNames = ['Januray', 'February', 'March', 'April', 'May', 'June',
+        'July', 'August', 'September', 'November', 'December'];
+
+    // explicit object enhancedDate   
+    var enhancedDate = {    
+
+        checkDate: function checkDate(userDate) {
+            if (theDate instanceof Date) {
+                return true; 
+            } else {
+                return false;    
+            }
+        },
+
+        setDate: function setDate(userDate) { 
+            if (userDate instanceof Date) {
+                theDate = userDate;
+            } else { 
+                if (typeof userDate === 'number') {
+                theDate = new Date(userDate);
+                } else {
+                    theDate = new Date();
+                }
+            }           
+            //console.log("In set date - date = ", theDate);      
+        }, 
+
+        getDate: function getDate(userDate) {            
```

Comment:
This function should return theDate if someone passes in true. If nothing, or false is passed in, it should return the millisecond representation of theDate.